### PR TITLE
Collections State

### PR DESCRIPTION
**Pull request by gilhsu**
New
- install `uuid` and `react-native-get-random-values` libraries to generate collection uuids
- `collections` state
- `selectedCollection` state
- `filteredResourceTypesSelector`, the 'god' object that consolidates the timeRange and resourceType filters - used to make `collectionFlattenedSubTypesSelector` and `selectedFlattenedSubTypesSelector`
- `collectionFlattenedSubTypesSelector`, outputs a flattened data model for rendering the subType accordions in `ContentPanel`
- `selectedFlattenedSubTypesSelector`, outputs a flattened data model for just the selected resourceType's subType accordions

Updated
- `ResourceCard`, include tappable checkmark icons to add/remove resource
- `SubTypeAccordion`, include tappable checkmark icons to add/remove all resources in that subType
- `SubTypeAccordionContainer`, pull out conditional logic to parent components
- `CatalogScreen`, conditional logic for rendering `SubTypeAccordionContainer` and take in data prop
- `ContentPanel`, include data props and a display all subtypes prop for `SubTypeAccordionContainer`

### CODE REVIEW

**Review by vadi2 — APPROVED** [review]

File: src/redux/selectors/index.js
Comment (foxtree) [inline src/redux/selectors/index.js:259]:
```
const { minimumDate, maximumDate } = timelineProps;
const { dateRangeStart = minimumDate, dateRangeEnd = maximumDate } = dateRangeFilterFilters;
```
and dispense w/ `activeDateStart` / `activeDateEnd` ?

File: src/redux/selectors/index.js
Comment (foxtree) [inline src/redux/selectors/index.js:291]: I wonder if it could be more clear w/ an intervening object assignment, and/or Object literal notation

```
const subtypeViewModel = acc[resourceType].subTypes[subType]

const subtypeViewModel = {
...subtypeViewModel,
  foo: bar, 
  baz: biff
}
```

File: src/redux/selectors/index.js
Comment (foxtree) [reply src/redux/selectors/index.js:259]: there is also a new `timelineRangeSelector`, that does ^this.

File: src/redux/selectors/index.js
Comment (foxtree) [inline src/redux/selectors/index.js:281]: there is already a selector for filtering all resources by date.  (and another that plucks relevant values for getting id and type)  this could be leveraged, but it would have to be the primary data Array this whole selector operates on, w/ grouping occurring, here

File: src/redux/selectors/index.js
Comment (foxtree) [reply src/redux/selectors/index.js:259]: or use `timelineItemsInRangeSelector` (see comment below) and modify `pickTimelineFields` to also pluck `subType`)  -- eliminating any other time-related selector fed to this one.

Comment (gilhsu) [reply src/redux/selectors/index.js:259]: This will have to be a refactor after the demo

File: src/redux/selectors/index.js
Comment (gilhsu) [reply src/redux/selectors/index.js:281]: This will have to be a refactor after the demo

File: src/redux/reducers/index.js
Comment (foxtree) [inline src/redux/reducers/index.js]: looks like two new constants were added to origin/master:
PLURAL_RESOURCE_TYPES and SINGULAR_RESOURCE_TYPES

**Comment by foxtree** [comment]
we could always merge, and fix/refactor/simplify, later...

I'm trying to test how this runs, now, after having rebased it on `origin/master`  (there were a number of conflicts w/ recently merged PRs)

File: src/redux/selectors/index.js
Comment (foxtree) [reply src/redux/selectors/index.js:259]: `timelineItemsInRangeSelector` is maybe the safest option. it tests whether there _is_ a start/end date, and otherwise returns an empty Array.

...This is an important check, b/c it is conceivable that particular UI views render before any records with dates are processed, and that would prevent an exception, here.

Comment (foxtree) [reply src/redux/selectors/index.js:259]: could derive a selector that combines type and date filters.   (probably more performant: type 1st, range testing, 2nd)

...then, all this thing does is group by type/subtype, and flag what's in the collection or selected.

File: src/redux/selectors/index.js
Comment (foxtree) [reply src/redux/selectors/index.js:281]: yes; I think there should be dedicated pass to trim / simplify the proliferating number of selectors in this file.

...that will be easier in the context of all functionality having been implemented.

**Comment by foxtree** [comment]
minor detail, that's not the focus of this PR --
I've been wondered whether the collapsed state arrow should point to the right:
image

some OS / UI idioms have icons indicate what the user action would "do",(rather than "is) --  and I think it is less ambiguous to have right + down than up + down


* EDIT: I guess the MUI (Android) default is up/down, so: maybe not.

**Review by foxtree — APPROVED** [review]
approved; we should thoroughly check after rebasing / merging

File: src/redux/selectors/index.js
Comment (foxtree) [reply src/redux/selectors/index.js:282]: b/c grouping requires an intervening representation as a hash, it might be simpler to sort (by label) in the UI, and only for expanded accordions.

**Review by vadi2 — APPROVED** [review]

**Pull request merged**
